Add private network mode

nprokopic opened the pull request.

marians commented:
@nprokopic Meanwhile the entire network section has moved in the schema into the `/connectivity` section. If this PR is going to be merged, it would be great if the additions would end up under `/connectivity`, too.

nprokopic commented:
> @nprokopic Meanwhile the entire network section has moved in the schema into the `/connectivity` section. If this PR is going to be merged, it would be great if the additions would end up under `/connectivity`, too.

I rebased my other PR and updated it according to the new schema, will do the same with this one.

marians commented:
@nprokopic Another change happened that affcets this PR. Yesterday we changed this repo to have the default values encoded in the `values.schema.json`. `values.yaml` gets generated from the schema using https://github.com/giantswarm/helm-values-gen
